docs/docs/testing-react-components.md

```diff
@@ -0,0 +1,57 @@
+---
+title: "Testing React components"
+---
+
+_The recommended testing framework is [Jest](https://jestjs.io/). This guide assumes that you followed the [Unit testing](docs/unit-testing) guide to setup Jest._
+
+Kent C. Dodds' [react-testing-library](https://github.com/kentcdodds/react-testing-library) has risen in popularity since its release and is a great replacement for [enzyme](https://github.com/airbnb/enzyme). It encourages you to query the DOM in the same way the user would. Hence the guiding principle:
+
+> The more your tests resemble the way your software is used, the more confidence they can give you.
+
+It provides light utility functions on top of `react-dom` and `react-dom/test-utils` and gives you the confidence that refactors of your component in regards to the implementation (but not functionality) don't break your tests.
+
+## Installation
+
+Install the library as one of your project's `devDependencies`. Optionally you may install `jest-dom` to use its [custom jest matchers](https://github.com/gnapse/jest-dom#custom-matchers).
+
+```sh
+npm install --save-dev react-testing-library jest-dom
+```
+
+Create the file `setup-test-env.js` at the root of your project. Insert this code into it:
+
+```js
+import 'jest-dom/extend-expect';
+
+// this is basically: afterEach(cleanup)
+import 'react-testing-library/cleanup-after-each';
+```
+
+This file gets run automatically by Jest before every test and therefore you don't need to add the imports to every single test file.
+
+Lastly you need to tell Jest where to find this file. Open your `package.json` and add this entry to your `"jest"` section:
+
+```json
+"jest": {
+  "setupTestFrameworkScriptFile": "<rootDir>/setup-test-env.js"
+}
+```
+
+## Usage
```

Comment:
Again, in which scenarios might someone use this code? What kinds of files would you want to use it?